src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorSourceUpdater.cs

```diff
@@ -56,10 +57,16 @@ public class RazorSourceUpdater : IUpdater<RazorCodeDocument>
         /// <param name="textMatcher">The text matching service to use for correlating old sections of text in Razor documents with updated texts.</param>
         /// <param name="textReplacer">The text replacing service to use for updating replaced texts in the Razor documents.</param>
         /// <param name="logger">An ILogger to log diagnostics.</param>
-        public RazorSourceUpdater(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<CodeFixProvider> codeFixProviders, ITextMatcher textMatcher, ITextReplacer textReplacer, ILogger<RazorSourceUpdater> logger)
+        public RazorSourceUpdater(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<CodeFixProvider> codeFixProviders, IEnumerable<AdditionalText> additionalTexts, ITextMatcher textMatcher, ITextReplacer textReplacer, ILogger<RazorSourceUpdater> logger)
```

Comment:
Why not pass an ImmutableArray<AdditionalText> here since that's what you actually want

Reply:
I'm using ctor injection here. I wasn't sure whether Microsoft.Extensions.DependencyInjection would build an ImmutableArray for me. I suppose that's easy enough to test, though. I'll try it.

Reply:
Unfortunately, it looks like we'll need to stick with IEnumerable because Autofact doesn't know how to supply ImmutableArray:

```
Autofac.Core.DependencyResolutionException: An exception was thrown while activating Microsoft.DotNet.UpgradeAssistant.Cli.ConsoleUpgrade -> Microsoft.DotNet.UpgradeAssistant.UpgraderManager -> Microsoft.DotNet.UpgradeAssistant.UpgradeStepOrderer -> ?:Microsoft.DotNet.UpgradeAssistant.UpgradeStep[] -> Microsoft.DotNet.UpgradeAssistant.Steps.Razor.RazorUpdaterStep -> ?:Microsoft.DotNet.UpgradeAssistant.IUpdater`1[[Microsoft.AspNetCore.Razor.Language.RazorCodeDocument, Microsoft.AspNetCore.Razor.Language, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]][] -> Microsoft.DotNet.UpgradeAssistant.Steps.Razor.RazorSourceUpdater.
 ---> Autofac.Core.DependencyResolutionException: None of the constructors found with 'Autofac.Core.Activators.Reflection.DefaultConstructorFinder' on type 'Microsoft.DotNet.UpgradeAssistant.Steps.Razor.RazorSourceUpdater' can be invoked with the available services and parameters:
Cannot resolve parameter 'System.Collections.Immutable.ImmutableArray`1[Microsoft.CodeAnalysis.AdditionalText] additionalTexts' of constructor 'Void .ctor(System.Collections.Generic.IEnumerable`1[Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer], System.Collections.Generic.IEnumerable`1[Microsoft.CodeAnalysis.CodeFixes.CodeFixProvider], System.Collections.Immutable.ImmutableArray`1[Microsoft.CodeAnalysis.AdditionalText], Microsoft.DotNet.UpgradeAssistant.Steps.Razor.ITextMatcher, Microsoft.DotNet.UpgradeAssistant.Steps.Razor.ITextReplacer, Microsoft.Extensions.Logging.ILogger`1[Microsoft.DotNet.UpgradeAssistant.Steps.Razor.RazorSourceUpdater])'.
   at Autofac.Core.Activators.Reflection.ReflectionActivator.GetAllBindings(ConstructorBinder[] availableConstructors, IComponentContext context, IEnumerable`1 parameters)
```

Reply:
Why are you getting it via dependency injection? Is the main one the WebTypeReplacements? Why not use the extension model to allow additional files that get added in?